extension/flink-doris-connector/src/main/java/org/apache/doris/flink/backend/BackendClient.java

```diff
@@ -0,0 +1,220 @@
+                    logger.warn("The status of open scanner result from {} is '{}', error message is: {}.",
+                            routing, result.getStatus().getStatus_code(), result.getStatus().getError_msgs());
+                    continue;
+                }
+                return result;
+            } catch (TException e) {
+                logger.warn("Open scanner from {} failed.", routing, e);
+                ex = e;
+            }
+        }
+        logger.error(ErrorMessages.CONNECT_FAILED_MESSAGE, routing);
+        throw new ConnectedFailedException(routing.toString(), ex);
+    }
+
+    /**
+     * get next row batch from Doris BE
+     * @param nextBatchParams thrift struct to required by request
+     * @return scan batch result
+     * @throws ConnectedFailedException throw if cannot connect to Doris BE
+     */
+    public TScanBatchResult getNext(TScanNextBatchParams nextBatchParams) throws DorisException {
+        logger.debug("GetNext to '{}', parameter is '{}'.", routing, nextBatchParams);
+        if (!isConnected) {
+            open();
+        }
+        TException ex = null;
+        TScanBatchResult result = null;
+        for (int attempt = 0; attempt < retries; ++attempt) {
+            logger.debug("Attempt {} to getNext {}.", attempt, routing);
+            try {
+                result  = client.get_next(nextBatchParams);
+                if (result == null) {
+                    logger.warn("GetNext result from {} is null.", routing);
+                    continue;
+                }
+                if (!TStatusCode.OK.equals(result.getStatus().getStatus_code())) {
+                    logger.warn("The status of get next result from {} is '{}', error message is: {}.",
+                            routing, result.getStatus().getStatus_code(), result.getStatus().getError_msgs());
+                    continue;
+                }
+                return result;
+            } catch (TException e) {
+                logger.warn("Get next from {} failed.", routing, e);
+                ex = e;
+            }
+        }
+        if (result != null && (TStatusCode.OK != (result.getStatus().getStatus_code()))) {
+            logger.error(ErrorMessages.DORIS_INTERNAL_FAIL_MESSAGE, routing, result.getStatus().getStatus_code(),
+                    result.getStatus().getError_msgs());
+            throw new DorisInternalException(routing.toString(), result.getStatus().getStatus_code(),
+                    result.getStatus().getError_msgs());
+        }
+        logger.error(ErrorMessages.CONNECT_FAILED_MESSAGE, routing);
+        throw new ConnectedFailedException(routing.toString(), ex);
+    }
+
+    /**
+     * close an scanner.
+     * @param closeParams thrift struct to required by request
+     */
+    public void closeScanner(TScanCloseParams closeParams) {
+        logger.debug("CloseScanner to '{}', parameter is '{}'.", routing, closeParams);
```

Comment:
CloseScanner from ? Close scanner from? CloseScanner to? Close scanner to ? Which is correct?